### PR TITLE
Discourage use of 'window' and 'document'

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This should help us find issues such as https://github.com/firebase/firebase-js-sdk/issues/2870 before they surface in ReactNative and WebWorkers.

### CODE REVIEW

File: packages/firestore/src/platform_browser/browser_platform.ts [hunk 1]
```diff
@@ -34,10 +35,14 @@ export class BrowserPlatform implements Platform {
   }
 
   get document(): Document | null {
+    // `document` is not always available, e.g. in ReactNative and WebWorkers.
+    // eslint-disable-next-line no-restricted-globals
     return typeof document !== 'undefined' ? document : null;
   }
 
   get window(): Window | null {
+    // `window` is not always available, e.g. in ReactNative and WebWorkers.
```
Comment (wilhuff) [inline packages/firestore/src/platform_browser/browser_platform.ts:44]: Drive-by comment: should this fall back on `self` in a web worker?

Comment (schmidt-sebastian) [reply packages/firestore/src/platform_browser/browser_platform.ts:44]: I don't know and we should be careful if we want to make this change. FWIW, this doc says to always use `self`: https://developer.mozilla.org/en-US/docs/Web/API/Window/self

This works because JS in browsers prefixes any unknown globals with `window` and `window.self` seems to be a thing.
